### DISCUSSION TITLE
Render lazy components in server?

### DISCUSSION THREAD

**Discussion by SeeThruHead** [vikejs/vike#1050] [Discussion]
### Description

We are looking to move from nextjs to vite. vite-plugin-ssr seems like the way to do it.
But when looking through the docs the dynamic import section is under the "client-only components section"

This won't work for us, we want to server render the html for dynamic imports (like how nextjs works)

Is there something I'm missing about this? Any plans for it?
Thanks

**Comment by brillout** [comment]
What's your use case? Why not importing the component as usual?

In principle, if Next.js allows this, then React must have a feature to enable this.

Have you tried to `React.lazy(() => import('my-component')))` without using an effect? In general, if you load the lazy component during SSR then it should be rendered to HTML just like any other component.

> "client-only components section"

I agree it's confusing. Let's continue the discussion and I'll update the docs then.

**Reply by brillout** [reply]
@straystreyk You basically want a React component that a) on the client-side is loaded and rendered after the initial page load/render, and b) on the server-side is rendered to HTML &mdash; am I guessing your use case right?

> Have you tried to `React.lazy(() => import('my-component')))` without using an effect? In general, if you load the lazy component during SSR then it should be rendered to HTML just like any other component.

Have you tried this?

A hydration mismatch seems inevitable here, so I'd go for https://legacy.reactjs.org/docs/dom-elements.html#suppresshydrationwarning.

Keep us updated.

@SeeThruHead Did you find a solution?

**Comment by straystreyk** [comment]
@brillout Hi. Decided not to create a new discussion and ask here, as the topic is very similar, if you help I will be very grateful.
I use the vite-ssr-plugin and when loading the main page of the site, I have chunks loaded that are not used on this page. I decided to use React.lazy - and it only works on the client side. When loading a page when SSR occurs, the application breaks because the html that comes from the server does not match the one that renders the UI. 
Is there a possibility in vite-ssr-plugin, load only those chunks that are needed on each of the pages. Or alternatively is it possible to connect @loadable/components to my vite react app which uses vite-ssr-plugin?

**Reply by brillout** [reply]
Have you tried the `<ClientOny>` implementation described at https://vite-plugin-ssr.com/client-only-components#react? Minimal reproduction welcome showcasing an issue with that implementation.

**Reply by straystreyk** [reply]
@brillout TY for answer. I've tried but it's not exactly what i need. As an example, I have an article page and the react-mardown library is used to process markdown. I don't want to use ClientOnly because then the page won't be SEO friendly. When I load the main page, I see how a chunk of react-markdown comes to me, which is used in articles but not on the main page, and I don’t need it. Are there other ways to get around this?

**Reply by brillout** [reply]
I have difficulties understanding your use case. I'm inclined to think it's a user land issue. That said, feel free to create a minimal reproduction showcasing that there is something about vite-plugin-ssr that is broken or that can be improved.

I'm hiding your comment as off-topic because it seems to be a wholly different use case. Feel free to create a new discussion.

**Edit:** Nevermind, I think I'm starting to understand what you want. See my sibling comment.
